emcc

```diff
@@ -750,6 +755,16 @@ try:
   # counter for the next index that should be used.
   next_arg_index = len(newargs)
 
+  if pthreads:
+    settings_changes.append('USE_PTHREADS=1')
+    if not any(s.startswith('PTHREAD_POOL_SIZE=') for s in settings_changes):
+      settings_changes.append('PTHREAD_POOL_SIZE=0')
+    js_libraries.append(shared.path_from_root('src', 'library_pthread.js'))
+  else:
+    js_libraries.append(shared.path_from_root('src', 'library_pthread_stub.js'))
```

Comment:
I'm not sure I follow this comment.

The usage will be like this: If one wants to use pthreads, one must use the compiler flag `-s USE_PTHREADS=1`, and the linker flag `-lpthread`. Only using one or the other is not enough, since the changes extend to bitcode level during compilation, and at link time as well.

If these flags are not specified, then the old versions of `pthread_xxx` are used, which are the minimal implementation/stubs. Some codebases might not be interested in actual pthreads support, but might still depend on the stub versions to compile. Also our test suite depends on this.

I will have to write up documentation for developers on how to target pthreads, but that will be a bit later.